pkg/cloud/vsphere/actuators/cluster/actuator.go

```diff
@@ -72,6 +77,28 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) (opErr error) {
 		return errors.Wrapf(err, "unable to reconcile certs while reconciling cluster %q", ctx)
 	}
 
+	// sync kubeconfig secrets for cluster
+	kubeConfig, err := a.GetKubeConfig(cluster, nil)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get kubeconfig for cluster %q", ctx)
+	}
+
+	secret := &apiv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-kubeconfig", cluster.Name),
+		},
+		StringData: map[string]string{
+			"value": kubeConfig,
+		},
+	}
+
+	_, err = a.coreClient.Secrets(cluster.Namespace).Create(secret)
```

Comment:
Hi @andrewsykim,

Please collapse into a single line since the non-error return value is ignored. In fact, since [`apierrors.IsAlreadyExist`](https://github.com/kubernetes/apimachinery/blob/d16981aedf33f88b7434bd3bdbcc2e1f940aecd0/pkg/api/errors/errors.go#L465-L468) calls [`apierrors.ReasonForError`](https://github.com/kubernetes/apimachinery/blob/d16981aedf33f88b7434bd3bdbcc2e1f940aecd0/pkg/api/errors/errors.go#L616-L623) and they can both accept a `nil` error, please rewrite like so:

```golang
	if _, err := a.coreClient.Secrets(cluster.Namespace).Create(secret); !apierrors.IsAlreadyExists(err) {
		return errors.Wrapf(err, "error creating kubeconfig secret for cluster %q", ctx)
	}
```

Reply:
Realizing though that `!apierrors.IsAlreadyExists(err)` returns true for `err == nil` in which case we don't want to return err.

Reply:
Ah, can still collapse as `err != nil && !apierrors.IsAlreadyExists(err)` :)